### DISCUSSION TITLE
Modbus RTU pin mapping

### DISCUSSION THREAD

**Discussion by fahadalisarwar1** [eModbus/eModbus#54] [Q&A] [answered]
I was just trying out the modbus RTU06 example and I noticed in the example, RX2=17, TX2=16, whereas the board that I have has these pins swapped. I have TX2=17, RX2=16. I amusing espressif dev kit. Can you confirm if the pins in the example are with respect to the esp32 or RS485 to TTL converter ?

**Comment by Miq1** [comment]
Yes, your pinout is correct for the DevkitC. I mentioned the swap somewhere as being advantageous for my own PCB layout. You may use almost any pins on the ESP32 anyway.

**Comment by bertmelis** [comment, marked as answer]
On the ESP32 you have a whole matrix at your disposal. The technical reference manual Chapter 5 explains the GPIO Matrix & IOMUX in detail: https://www.espressif.com/sites/default/files/documentation/esp32_technical_reference_manual_en.pdf

But the Arduino framework makes it easy by allowing the pins of your choice to be passed as arguments in `Serial.begin(...)`.
